src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,559 @@
+I'll explore in this article different techniques you can use to accelerate
+your media playback by actively preloading resources depending on your use
+case.
+
+Note: Unless specified otherwise, this article also applies to the audio element.
+
+### TL;DR {: .hide-from-toc }
+
+<table>
+  <tbody>
+    <tr>
+      <th></th>
+      <th>
+Pros
+      </th>
+      <th>
+Cons
+      </th>
+    </tr>
+    <tr>
+      <td rowspan=3 style="white-space: nowrap">
+<a href="#video_preload_attribute">Video preload attribute</a>
+      </td>
+      <td rowspan=3>
+Simple to use for an unique file hosted on a web server
+      </td>
+      <td>
+Browser may completely ignore the attribute
+      </td>
+    <tr>
+      <td>
+Resource fetching starts when HTML document has been completely loaded and
+parsed
+      </td>
+    </tr>
+    <tr>
+      <td>
+Doesn't work with MSE
+      </td>
+    </tr>
+    <tr>
+      <td rowspan=3 style="white-space: nowrap">
+<a href="#link_preload">Link preload</a>
+      </td>
+      <td>
+Forces the browser to make a request for a video resource without blocking
+the document's <code>onload</code> event
+      </td>
+      <td rowspan=3>
+Doesn't work with HTTP Range requests
+      </td>
+    <tr>
+      <td>
+Compatible with MSE and segment files
+      </td>
+    </tr>
+    <tr>
+      <td>
+Should be used only for audio files or short video clips when fetching full
+resource
+      </td>
+    </tr>
+    <tr>
+      <td>
+<a href="#manual_buffering">Manual buffering</a>
+      </td>
+      <td>
+Full control
+      </td>
+      <td>
+Complex error handling is website responsibility
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+## Video preload attribute
+
+If the video source is an unique file hosted on a web server, you may want to
+use the video `preload` attribute to provide a hint to the browser as to [how
+much information or content to preload]. This means [Media Source Extensions
+(MSE)] is not compatible with `preload`.
+
+Resource fetching will start only when initial HTML document has been
+completely loaded and parsed (eg. `DOMContentLoaded` event) while the very
+different `load` event will be fired when resource has actually been fetched.
+
+<figure class="TODO">
+  <img src="/web/fundamentals/media/images/video-preload/video-preload.svg">
+</figure>
+
+Setting `preload` value to `metadata` indicates that the user is not expected
+to need the video, but that fetching its metadata (dimensions, track list,
+duration, and so on) is desirable.
+
+```
+<video id="video" preload="metadata" src="file.mp4" controls></video>
+
+<script>
+  video.addEventListener('progress', function() {
+    if (video.buffered.length === 0) return;
+
+    var bufferedSeconds = video.buffered.end(0) - video.buffered.start(0);
+    console.log(bufferedSeconds + ' seconds of video are ready to play!');
+  });
+</script>
+```
+
+With `preload` value set to `auto`, we're saying th at downloading the entire
```

Comment:
> I'd also add that while the browser may ignore preload=auto, I don't think anyone ignores preload=metadata or preload=none.

Browser ignores `preload=metadata` in several conditions: Chrome Data Saver and Android 4.3.

> s/th at/that/
 
Fixed.

> I also wouldn't use the words "the entire video". I don't think any browser does that right away, they only cache enough such that canplaythrough is satisfied.

Fixed.